control/vector_field_waypoint.hpp

```diff
@@ -56,7 +57,7 @@ extern "C"
  */
 typedef struct
 {
-    const Mavlink_waypoint_handler*   waypoint_handler;         ///< Waypoint list (input)
+    Mavlink_waypoint_handler*   waypoint_handler;         ///< Waypoint list (input)
```

Comment:
answer: that is because ```Mavlink_waypoint_handler::waypoint_from_index(int i)``` is not marked const.
Indeed, it creates a home waypoint if the waypoint list is empty. The methods `waypoint_from_index`, `next_waypoint` and `current_waypoint` can be marked const by adding a home waypoint that would be returned if the waypoint list is empty.